units/src/main/java/org/apache/tuweni/units/bigints/BaseUInt256Value.java

```diff
@@ -338,12 +338,12 @@ public boolean equals(Object obj) {
     if (obj == this) {
       return true;
     }
-    if (obj instanceof Bytes) {
-      final Bytes other = (Bytes) obj;
-      return this.value.equals(other);
+    if (!(obj instanceof Bytes32)) {
+      return false;
     }
 
-    return false;
+    Bytes32 other = (Bytes32) obj;
```

Comment:
IMO, you don't need to cast, you just delegate the work to UInt256. I wonder if it is not even better to do directly
return this.value.equals(other); as the UInt256 implementation does the same checks.